megamek/src/megamek/client/ui/swing/GUIPreferences.java

```diff
@@ -1560,6 +1564,69 @@ public LabelDisplayStyle getUnitLabelStyle() {
         }
     }
 
+    /***
+     * @return The color associated with this movement type
+     */
+    public Color getColorForMovement(EntityMovementType movementType) {
+        switch (movementType) {
+            case MOVE_RUN:
+            case MOVE_VTOL_RUN:
+            case MOVE_OVER_THRUST:
+                return getColor("AdvancedMoveRunColor");
+            case MOVE_JUMP:
+                return getColor("AdvancedMoveJumpColor");
+            case MOVE_SPRINT:
+            case MOVE_VTOL_SPRINT:
+                return getColor("AdvancedMoveSprintColor");
+            case MOVE_ILLEGAL:
+                return getColor("AdvancedMoveIllegalColor");
+            default:
+                return getColor("AdvancedMoveDefaultColor");
+        }
+    }
+
+    /***
+     * @return The color associated with a movement type
+     */
+    public Color getColorForMovement(EntityMovementType movementType, boolean isMASCOrSuperCharger, boolean isBackwards) {
+        if (isMASCOrSuperCharger)
+        {
+            return getColor("AdvancedMoveMASCColor");
+        }
+
+        if (isBackwards)
+        {
+            return getColor("AdvancedMoveBackColor");
+        }
+
+        return getColorForMovement(movementType);
+    }
+    /***
+     * @return The color associated with a heat in the range 0-30
+     */
+    public Color getColorForHeat(int heat) {
+        return getColorForHeat(heat, Color.LIGHT_GRAY);
+    }
+
+    /***
+     * @return The color associated with a heat in the range 0-30
+     */
+    public Color getColorForHeat(int heat, Color defaultColor) {
+        if (heat <= 5) {
+            return defaultColor;
+        }
+        if (heat < 10) {
+            return Color.GREEN;
+        }
+        if (heat < 20) {
+            return Color.YELLOW;
+        }
+        if (heat < 30) {
+            return Color.ORANGE;
+        }
+        return Color.RED;
```

Comment:
I'd recommend using if - else if - else instead of a bunch of if statements to improve readability going forward.